include/zephyr/net/wifi_mgmt.h

```diff
@@ -116,14 +117,18 @@ struct wifi_status {
 typedef void (*scan_result_cb_t)(struct net_if *iface, int status,
 				 struct wifi_scan_result *entry);
 
-struct net_wifi_mgmt_offload {
+struct net_wifi_mgmt {
 	/**
 	 * Mandatory to get in first position.
 	 * A network device should indeed provide a pointer on such
 	 * net_if_api structure. So we make current structure pointer
 	 * that can be casted to a net_if_api structure pointer.
 	 */
-	struct net_if_api iface_api;
+#ifdef CONFIG_WIFI_OFFLOAD
+	struct net_if_api wifi_iface;
+#else
+	struct ethernet_api wifi_iface;
+#endif
```

Comment:
It is not that CONFIG_WIFI_OFFLOAD are anything special. It is ESP32 that operates on Ethernet and requires to embed `struct ethernet_api`.

Reply:
You can probably use union here.